### PR TITLE
Redis: don't record values that are written to the DB

### PR DESCRIPTION
**Pull request by owais**
Redis instrumentation now skips recording actual values written to the
database for known write commands.

### CODE REVIEW

File: contrib/go-redis/redis/redis.go [hunk 1]
```diff
@@ -193,7 +191,40 @@ func createWrapperFromClient(tc *Client) func(oldProcess func(cmd redis.Cmder) e
...
+	parts := strings.Split(strings.Trim(cmdStr, " "), " ")
+	if len(parts) == 0 {
+		return cmdStr, cmdStr, 0
+	}
+
+	cmd := parts[0]
+	argsLength := len(parts) - 1
+	numArgs, ok := cmdCaptureArguments[cmd]
+	if !ok {
+		return cmdStr, cmd, argsLength
```
Comment (johnbley) [inline contrib/go-redis/redis/redis.go:213]: If I understand this logic correctly, any unrecognized/not-explicitly-encoded command will just have part[0] (the command name and only the command name, no args) as the span name?

Comment (owais) [reply contrib/go-redis/redis/redis.go:213]: It's the reverse. Any unrecognized command will use the full command with args as the tag value (preserve current behavior).

Comment (johnbley) [reply contrib/go-redis/redis/redis.go:213]: Perhaps for safety it should be the other way?  The Jedis (Java) instrumentation appears to just use the command name (first part) and nothing else (though I like the idea of porting the "this number of params is safe" there).

Comment (owais) [reply contrib/go-redis/redis/redis.go:213]: The list of known commands is quite exhaustive. I actually went over all commands and listed down every command I found that wrote data to the DB or queried data using values that might be in the DB. All other commands either use keys/IDs and never take any "data" as arguments so it's safe to keep them at least from data volume perspective. The downside is that a new "write" command in future would escape this filter but it's probably not that big of a deal given than this PR mainly tries to reduce trace volume.

**Comment by owais** [comment]
Merging this but please feel free to review and provide any addition comments @johnbley @pjanotti. I'll fix in a follow up PR.

**Pull request merged**